crates/accelerate/src/commutation_checker.rs

```diff
@@ -294,20 +281,20 @@ impl CommutationChecker {
         qargs2: &[Qubit],
         cargs2: &[Clbit],
         max_num_qubits: u32,
+        approximation_degree: f64,
     ) -> PyResult<bool> {
-        // relative and absolute tolerance used to (1) check whether rotation gates commute
-        // trivially (i.e. the rotation angle is so small we assume it commutes) and (2) define
-        // comparison for the matrix-based commutation checks
-        let rtol = 1e-5;
-        let atol = 1e-8;
+        // If the average gate infidelity is below this tolerance, they commute. The tolerance
+        // is set to max(1e-12, 1 - approximation_degree), to account for roundoffs and for
+        // consistency with other places in Qiskit.
+        let tol = 1e-12_f64.max(1. - approximation_degree);
```

Comment:
This does mean we can't approximate more than what a 1e-12  tolerance provides. But it's the same logic we use in other places so I think that's fine.

Reply:
Yeah it is unfortunately. I don't like this very much either, but currently `approximation_degree` is `1.` everywhere per default which doesn't really allow us to set another default like `1-1e-12`. The other option would've been to use `None` as indicator, but that already means that the `target` error rates should be used.